styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -183,6 +190,20 @@
     this.forkJoinPool = new ForkJoinPool(REQUEST_CONCURRENCY);
   }
 
+  void indexActiveWorkflowInstances() {
+    // XXX: this listing is not strongly consistent
+    final List<Entity> indexEntries = StreamUtil.stream(
+        datastore.run(Query.newKeyQueryBuilder().setKind(KIND_ACTIVE_WORKFLOW_INSTANCE).build()))
+        .map(Key::getName)
+        .map(wfiKey -> activeWorkflowInstanceIndexShardEntryKey(datastore.newKeyFactory(), wfiKey))
```

Comment:
This will create the ancestor entry if it doesn't exist already, right?

Reply:
actually a comment of the line below

Reply:
no, the ancestor isn't needed, see comment and docs

Reply:
OK. As long as it is safe to concurrently store two children referring to the same non-exist parent. Didn't find that in doc.

Reply:
Yeah, that should be fine. The parent is not created and the docs specify that an entity ancestor does not have to exist. So there's no collision on implicitly creating a parent.